static/app/views/insights/pages/agents/layout.tsx

```diff
@@ -0,0 +1,18 @@
+import {Fragment} from 'react';
+import {Outlet, useMatches} from 'react-router-dom';
+
+import {AgentsPageHeader} from 'sentry/views/insights/pages/agents/agentsPageHeader';
+import {ModuleName} from 'sentry/views/insights/types';
+
+function AgentsLayout() {
+  const handle = useMatches().at(-1)?.handle as {module?: ModuleName} | undefined;
+
+  return (
+    <Fragment>
+      {handle && 'module' in handle ? <AgentsPageHeader module={handle.module} /> : null}
```

Comment:
note: I left out passing `headerTitle={<Fragment>{getAIModuleTitle(organization)}</Fragment>}` because `headerTitle` just falls back to `domainTitle` if it’s not present, and that does the exact same thing:

https://github.com/getsentry/sentry/blob/38fa875f734b7375cf3c3ca7ed17bb2559986ad8/static/app/views/insights/pages/agents/agentsPageHeader.tsx#L41-L42